src/Dataset.hs

```diff
@@ -0,0 +1,30 @@
+{-# LANGUAGE DeriveGeneric #-}
+
+module Dataset where
+
+import qualified Data.ByteString.Lazy as DBL
+import qualified Data.Serialize as DS (decodeLazy, encodeLazy, Serialize)
+import qualified Data.Vector as DV (length, Vector(..))
+import qualified Data.Vector.Serialize as DVS (genericGetVector, genericPutVector)
+import qualified GHC.Generics as GG (Generic(..))
+
+data Dataset a = Dataset {
+    datapoints :: [DV.Vector a],
+    targets :: [DV.Vector a]
+} deriving (Eq, GG.Generic, Show)
+
+instance (DS.Serialize a) => DS.Serialize (Dataset a)
+
+{-normalize :: (RealFloat a) => Dataset a -> Dataset a
+normalize dataset = ...-}
+
+fromByteString :: (DS.Serialize a) => DBL.ByteString -> Either String (Dataset a)
+fromByteString = DS.decodeLazy
+
+fromLists :: [DV.Vector a] -> [DV.Vector a] -> Either String (Dataset a)
+fromLists datapoints targets
+    | length datapoints /= length targets = Left "Mismatching dimensions between datapoints and targets."
+    | otherwise = Right $ Dataset datapoints targets
+
+toByteString :: (DS.Serialize a) => Dataset a -> DBL.ByteString
+toByteString = DS.encodeLazy
```

Comment:
do we need this ? can't we juste use `encodeLazy` when it's needed  ? (also the laziness part is lost in the renaming :p)

Reply:
At first I was also reading the file, but the IO monad was getting annoying. I left those 2 functions because they prevent the burden of importing Data.Serialize, Data.Vector, Data.Vector.Serialize and GHC.Generics in other places we might want to load / save a dataset.